swarm/src/lib.rs

```diff
@@ -2404,4 +2469,236 @@ mod tests {
         // Unfortunately, we have some "empty" errors that lead to multiple colons without text but that is the best we can do.
         assert_eq!("Failed to negotiate transport protocol(s): [(/ip4/127.0.0.1/tcp/80: : No listener on the given port.)]", string)
     }
+
+    #[cfg(test)]
+    mod port_use {
+        use std::str::FromStr;
+
+        use libp2p_core::multiaddr::Protocol;
+        use libp2p_core::multihash::Multihash;
+        use libp2p_identity::PeerId;
+
+        use crate::any_with_same_transport;
+        use crate::Multiaddr;
+
+        macro_rules! test_dial_with_listener {
+        ($test_name:ident, $listen_addrs:expr, $combination:expr, $test_explanation:expr) => {
+            #[test]
+            fn $test_name() {
+                let test_explanation = $test_explanation;
+                let listen_addrs: Vec<Multiaddr> = $listen_addrs.into_iter().map(|e| e.parse().unwrap()).collect();
+                $combination.into_iter().map(|(dial_addr_str, expected_output)| (dial_addr_str.parse::<Multiaddr>().unwrap(), expected_output)).for_each(|(dial_addr, expected_output)| {
+                    let should_port_reuse = any_with_same_transport(listen_addrs.iter(), &dial_addr);
+                    if expected_output {
+                        assert!(should_port_reuse, "Port should have been reused, since there exists at least one listener in {listen_addrs:?} for {dial_addr:?}. Details: {test_explanation}")
+                    } else {
+                        assert!(!should_port_reuse, "Port should not have been reused, since there doesn't exist a listener in {listen_addrs:?} for {dial_addr:?}. Details: {test_explanation}");
+                    }
+                })
+            }
+        };
+    }
+
+        const BASIC_LISTEN_ADDRS: [&str; 7] = [
+            "/ip4/127.0.0.1",
+            "/ip6/::1",
+            "/ip4/127.0.0.1/tcp/800/tls/ws",
+            "/ip4/127.0.0.1/udp/443/quic",
+            "/ip4/127.0.0.1/udp/443/quic-v1",
+            "/dns4/bootstrap.libp2p.io/tcp/80/tls/ws/tls",
+            "/dns6/bootstrap.libp2p.io/tcp/443/tls/ws/tls",
+        ];
+
+        test_dial_with_listener!(
+        reuse_if_dial_to_self,
+        BASIC_LISTEN_ADDRS,
+        BASIC_LISTEN_ADDRS.map(|addr| (addr, true)),
+        "If this test fails there is likely a major problem, because we won't reuse ports to dial ourselves."
```

Comment:
```suggestion
	        reuse_if_dial_to_self,
	        BASIC_LISTEN_ADDRS,
	        BASIC_LISTEN_ADDRS.map(|addr| (addr, true)),
	        "If this test fails there is likely a major problem, because we won't reuse ports to dial ourselves."
```

I think it would be easier to read if we'd ident this.

Reply:
You are right, it's just the formatting made by rustfmt.